macrosynergy/panel/make_zn_scores.py

```diff
@@ -26,50 +28,57 @@ def pan_neutral(df: pd.DataFrame, neutral: str = 'zero', sequential: bool = Fals
-        is set to True, will be computed using the available cross-sections on the
-        respective date. The code will not adjust for an incomplete set of cross-sections
-        on each date. For instance, if the first 100 days only have 3 cross-sections with
-        realised values out of the 4 defined, the rolling mean will be calculated using
-        the available subset.
+    Compute neutral values of return series based on a panel.
 
     :param <pd.Dataframe> df: "wide" dataframe with time index and cross section columns.
+        It must have the frequency at which values are re-estimated.
     :param <str> neutral: method to determine neutral level. Default is 'zero'.
         Alternatives are 'mean' and "median".
     :param <bool> sequential: if True (default) score parameters (neutral level and
         standard deviations) are estimated sequentially with cumulative concurrently
         available information only. If False one neutral value will be calculated for
         the whole panel.
-    :param <int> min_obs:
-    :param <bool> iis:
+    :param <int> min_obs: minimum required observations in days.
+    :param <bool> iis: if set to True, the number of days outlined by
+    "min_obs" will be calculated in sample (single neutral value for the time period)
+    whilst the remaining days are calculated on a rolling basis. However, if False, and
+    "sequential" equals True, the rolling neutral value will be calculated from the start
+    date.
 
     :return <np.ndarray> arr_neutral: row-wise neutral statistic. A single value produced
         per row. Therefore, a one-dimensional array will be returned whose length matches
         the row dimension of the dataframe df.
+
+
+    NB: It is worth noting that the evolving neutral level, if the "sequential" parameter
+        is set to True, will be computed using the available cross-sections on the
+        respective date. The code will not adjust for an incomplete set of cross-sections
+        on each date. For instance, if the first 100 days only have 3 cross-sections with
+        realised values out of the 4 defined, the rolling mean will be calculated using
+        the available subset.
     """
     no_rows = df.shape[0]
 
     if neutral == 'mean':
         if sequential and not iis:
             ar_neutral = rolling_mean_with_nan(dfw=df)
-
+            # Todo: incorrect, use pandas df.expanding.mean()
             ar_neutral[:min_obs] = np.nan
+            # Todo: incorrect as min_obs is defined on days, while dfw could be weeks
+            #  or months.
+            # Todo: calculate earliest date based on days (timedelta) and use
 
         elif sequential and iis:
             iis_neutral = np.repeat(df.iloc[0:min_obs].stack().mean(),
                                     min_obs)
 
             os_neutral = rolling_mean_with_nan(dfw=df)
+            # Todo: incorrect, use pandas df.expanding.mean()
```

Comment:
Why incorrect ? I'm not sure df.expanding.mean() works when computing a rolling mean across a panel.